### PR TITLE
Action Icons Redesign

### PR DESCRIPTION
**Pull request by Fatih20**
I'm a user of this cool app on Linux and upon using it I found that the icons here are very unpleasant. I'm no professional by any means but I have spent the last few months contributing icons for other open source projects.

This is certainly not my final proposal, I wanted to listen to feedback first before continuing the redesign of all the action icon. There are some icon that haven't been redesigned because of that.

Feel free to ask me why I made certain icon the way they are if you'd like to gain some insight.

### CODE REVIEW

**Comment by Rmano** [comment]
It would be nice to have a screenshot of the new icons here... ;-)

**Comment by Fatih20** [comment]
I haven't taken times to compile the software, I'll get on it later.

**Comment by Fatih20** [comment]
Screenshot from 2020-02-07 21 30 23
Screenshot from 2020-02-07 21 29 48
Screenshot from 2020-02-07 21 29 15

**Comment by Technius** [comment]
Interesting work! A few comments:
* In my opinion, the eraser looks like a magnet.
* The latex icon is an upgrade, but it might look better in a serif font.
* Not sure if this was intentional, but all of the file permissions were changed.
* The new Default tool icon looks like an update icon. Not sure how to compare with the current one though, since the current one looks pretty confusing (in my opinion).
* The redesigned Hand Tool icon is a pointer, but that suggests selection rather than panning.
* How well the icons look is going to depend a lot on the icon theme. For example, I use the Papyrus icon theme and I think the current minimalist look goes really well with it, but the redesign clashes in style.
* I just realized that we don't need custom icons for the audio player, since the [XDG desktop icon specification](https://specifications.freedesktop.org/icon-naming-spec/icon-naming-spec-latest.html) has the `media-*` icons.

**Comment by Fatih20** [comment]
If XDG already has it, why is the current version use its own icon? 
Yes, I agree the current one is shit and probably is just taken from an obscure free stock vector website. 
I'd also like to add that while the current version is minimalist like Papyrus, it is far from the quality of Papyrus. 
I'll look into your other concern regarding the individual icon.

**Comment by nicegamer7** [comment]
I have some minor notes:
- When I think of a highlighter, I usually think yellow
- The hand tool is more red than I'd expect

I really like them! Good job.

**Comment by Rmano** [comment]
Definitely a nice step forward, I like them.

**Comment by Fatih20** [comment]
Cool. What would it takes for this to be merged?

**Comment by Technius** [comment]
Sorry for the late response, I'll take another look at this when I have time tomorrow.

**Comment by Technius** [comment]
I was very busy this week, but I finally had the time to review today.
* Please revert the file permissions on the icons to `644`.
* The rotation snapping icon doesn't look right; the rotation icon blends in too much with the magnet. This will be problematic to people who have some color blindness.
* The grid snapping icon isn't descriptive enough--there needs to be some indicator that it concerns the grid.
* Why was the color of the icons in the sidebar changed?
* The new presentation mode icon is very nice.
* The various draw mode tools (e.g. arrow, line, etc.) have too much line thickness. For example, it is hard to see the endpoints of the Draw Line icon or the arrowhead of the Draw Arrow icon.
* Vertical Space icon looks too much like a select icon, but it does something much different.

**Comment by Fatih20** [comment]
Regarding the grid snapping, the reason I didn't make it about grid is because it should've been named just "snapping" since it works without the grid being shown/activated. By the way, the rotation snapping icon should've been renamed from "snapping" to "rotation snapping", because it only applies rotation snapping.
The color of the sidebar icon is now red because I'd like to give similar theme to icons that are placed in smilar location and has function concerning the same thing. Red is just a good color because it can do the various shading and whatnot that I wanted to do on the sidebar icon.
Vertical space icon looked like the select icon because in the current version, it also looked a lot like the select icon.

I'll look into your other criticism and fix it.

**Comment by Fatih20** [comment]
image

**Comment by LittleHuba** [comment]
Adding a new page and creating a new document look too similar. I would change the create a document icon to a page with a yellow or golden star (*). That would remove the possibility of confusing both. Additionally the star is normally the sign used for these types of icons where you create a new file.

**Comment by LittleHuba** [comment]
Also all the different colors make the design extremely unrestful and (sorry to phrase it that way) childlike. I'd stick to a small pallet of colors and make the icons mostly monochrome. This also simplifies compatibility with different themes users might choose.

The icons themselves look amazing though! I like the new rotation snapping!

**Comment by Fatih20** [comment]
Creating a new document and adding page icon isn't part of this redesign. Creating a new document icon is covered by the [XDG desktop icon specification](https://specifications.freedesktop.org/icon-naming-spec/icon-naming-spec-latest.html) not from the xournalpp, I might redesign adding page to differentiate it though.

I could design 2 set of icon, one monochrome and one colourful, and then the user themselves could choose which icon they'd prefer. But that requires a coder putting the options in the preferences.

Maybe a poll like the one for infinite pages if I could design one? I don't know tho, it's up to the maintainer.

**Comment by nicegamer7** [comment]
Personally I like the coloured icons, so an option to pick between coloured and monochrome would be preferable.

**Comment by Fatih20** [comment]
Any further inputs before merging this? @Technius

**Comment by Technius** [comment]
This is mostly ready for merge, but I have to agree with @nicegamer7 in that having a selectable colored/monochrome theme would be nice. Here's a simple plan to do this:
1. You'll have to put the new (colored) icons in a different folder, e.g. `ui/iconsNew`.
2. We'll merge in the changes into a separate branch.
3. From there, we'll add code in the settings that will allow users to select between the colored icons and the monochrome icons.
4. Once we have done so, we'll merge the new icon branch back into `master`.

Does this sound good? @LittleHuba @Fatih20

**Comment by Fatih20** [comment]
Sounds like it, but I'll prefer if we get rid of the old icon entirely. I could design the monochrome icon that you asked for.

**Comment by Fatih20** [comment]
image
Alright so here's how it's going to be. Monochrome in IconsMono and Colorful in IconsColor. Now, there's this IconsDark that IMO is unnecessary since the colorful icon works with both light and dark mode and the monochrome icon I'm making will follow. I'm going to get rid of it since it seems to be obsolete unless you think otherwise. @Technius

**Comment by Fatih20** [comment]
image
This is the first draft of the monochrome icons, any opinions? @Technius @LittleHuba 
This is the color palette I use.
image

**Comment by Technius** [comment]
That looks great! Just one minor issue: the `Fine` stroke is just a black dot, which may not be visible on dark themes (this is how it looks currently, but I think we should improve it if we can).

> Now, there's this IconsDark that IMO is unnecessary since the colorful icon works with both light and dark mode and the monochrome icon I'm making will follow. I'm going to get rid of it since it seems to be obsolete unless you think otherwise.

Yeah, we probably won't need `iconsDark` anymore.

**Comment by Fatih20** [comment]
Just curious, what year was it when the original icon got implemented?

**Comment by Technius** [comment]
From a quick git blame, 2018 in #276 and `fc70b1fc`.

**Comment by Fatih20** [comment]
image
Where can I access those colored icons and the black rectangle and circle? I'd like to modify them.

**Comment by Technius** [comment]
The color tools are drawn programmatically. If you create a mockup, we could try implementing the mockup in a followup PR.

**Comment by Fatih20** [comment]
There are some icon here that I can't actually found in the apps? Like the centerPage and Goto. Where are they?

**Comment by Fatih20** [comment]
Also, I think the icon naming scheme could be improved. The current one is really all over the place. But to change it, you'd have to rewrite the code that handles the UI.

**Comment by Fatih20** [comment]
Why is there a conflict here? I have gotten rid of iconDark. @Technius

**Comment by Technius** [comment]
Conflict due to #1765. Just override it with your changes when you merge.

**Comment by Fatih20** [comment]
WDYM?

**Comment by badshah400** [comment]
@Technius asked my opinion on the suggested new icons, so here goes.

In general, the icons look great @Fatih20 ! The coloured icons will probably work well with both light and dark themes. However, several of the monochrome icons seem to have really low contrast against dark theme variants. I am looking especially at the "Text", "Add/Edit TeX", and the "Toggle Fullscreen" button icons. To be fair, with the monochrome icons, I don't know if you can improve the contrast against the dark theme without spoiling it for lighter themes and vice versa, unless one uses symbolic icons.

The "Default Tool" icon kind of looks like a "Refresh"/"Reload" icon.

To be consistent with the monochrome look of icons, would it not be nice to have the "Undo"/"Redo" buttons, the "Fill" button, and the "Cut"/"Copy"/"Paste" buttons in monochrome too?

Thanks!

**Comment by y2kmurali** [comment]
Not sure if this is in scope for this specific PR (it's something that's been bothering me for a while) - the save icon looks awfully like a download button; is there any reason we don't use a more traditional save icon (i.e. a floppy disk)? Let me know if I should submit this in a separate issue.

**Comment by Fatih20** [comment]
@y2kmurali No. The save icon is supplied by the OS, not xournal.

**Comment by Fatih20** [comment]
I would like to also add that for future releases, IMHO it's better if action-icons used within the app (`ui/iconsMono` and `ui/iconsColor`) should be separated from the app icon in  `ui/pixmaps`.

**Comment by Technius** [comment]
> Not sure if this is in scope for this specific PR (it's something that's been bothering me for a while) - the save icon looks awfully like a download button; is there any reason we don't use a more traditional save icon (i.e. a floppy disk)?

The save/load icon comes from your icon theme. For example, Papirus uses a download icon as its  `document-save` icon.

> I would like to also add that for future releases, IMHO it's better if action-icons used within the app (`ui/iconsMono` and `ui/iconsColor`) should be separated from the app icon in `ui/pixmaps`. 

I'm not sure what mean, since the app icon is already in a different folder than the action icons.

**Comment by Fatih20** [comment]
Yes, but the actual app icon is technically not part of the ui and as such should be put in different folder altogether.

**Comment by Fatih20** [comment]
@badshah400 
Yes, the monochrome icon which doesn't work under dark theme is currently being reworked.
The default icon also have been reworked.

Latest revision will be posted as soon as they're finished.

As for the fill, copy, cut, and paste icon, they're supplied by the user's icon theme because they're in XDG desktop icon specification. While it's possible to override them, @Technius preferred if we use their icon theme instead when possible as shown below.
>     * I just realized that we don't need custom icons for the audio player, since the [XDG desktop icon specification](https://specifications.freedesktop.org/icon-naming-spec/icon-naming-spec-latest.html) has the `media-*` icons.

Thank you for the compliment!

**Comment by Fatih20** [comment]
Monochrome
Screenshot from 2020-03-30 17 55 18
Colored
Screenshot from 2020-03-30 17 54 49
Judge for yourself whether or not if this needs revision but don't suddenly merge this because I think there are still some things that needs to be changed.

**Comment by Fatih20** [comment]
By the way, I have made an issue regarding the palettes. https://github.com/xournalpp/xournalpp/issues/1812#issue-590320611

**Comment by jvwowd** [comment]
Since code modifications are being made anyway, would it be possible to have an option to choose between using system icons or X++ supplied icons for things like save, copy/paste, new, etc.? I personally would prefer a consistency in the application rather than partial consistency with my system.

Currently, the monochrome theme feels like it clashes with the system icons, and I imagine this could be much worse if using a different system theme (ditto for the colored theme).

**Comment by Fatih20** [comment]
@Technius Interesting questions above. We could also make an option to override the desktop icon theme in the settings.

**Comment by Fatih20** [comment]
There have been precedent of overriding the desktop icon theme such as mypaint so I think I'll go ahead and make the necessary icon. IMHO, there should be an option to override or to not override. @Technius

**Comment by Fatih20** [comment]
#1880

**Comment by Fatih20** [comment]
It seems like I would have to do a minor tweaks to all of the icons I've made so there will be major setback.

**Comment by LittleHuba** [comment]
This already looks amazing!

Please also design icons replacing the ones that are currently handled by the OS icon theme. We should not rely on those anymore as they lead to some awful design glitches. The original thought there was that people rely on same looking icons to find the actions they search for. But to tell the truth, having a matching icon set of descriptive icons is the better way to go there in my opinion. We'll just have to stick to the common code there. A floppy for saving, page with a star for creating a new document,...

We'll switch the code loading those icons in a follow up PR but would need the appropriate icons in this one to make it smooth.

**Comment by Fatih20** [comment]
> This already looks amazing!
> 
> Please also design icons replacing the ones that are currently handled by the OS icon theme. We should not rely on those anymore as they lead to some awful design glitches. The original thought there was that people rely on same looking icons to find the actions they search for. But to tell the truth, having a matching icon set of descriptive icons is the better way to go there in my opinion. We'll just have to stick to the common code there. A floppy for saving, page with a star for creating a new document,...
> 
> We'll switch the code loading those icons in a follow up PR but would need the appropriate icons in this one to make it smooth.

Okay, I'll start work on the icon handled by the OS.

**Comment by Fatih20** [comment]
@LittleHuba What if the save OS handled icon that I made aren't conventional. E.g. not using a floppy disk for save, not using page with a star for new document. ?

**Comment by LittleHuba** [comment]
I'd say it depends on how they look. If they are easily recognizable it is okay otherwise you should stick with the concept of the OS icons. 

Could you post them here?

**Comment by Fatih20** [comment]
> I'd say it depends on how they look. If they are easily recognizable it is okay otherwise you should stick with the concept of the OS icons.
> 
> Could you post them here?

They're already here. Look at the icons in this commit named `document-x`.

**Comment by LittleHuba** [comment]
Sorry can't find them. Are you sure you included them in the commit and pushed it?

**Comment by Fatih20** [comment]
> Sorry can't find them. Are you sure you included them in the commit and pushed it?

Sorry, you can find them here.

**Comment by LittleHuba** [comment]
So I'm happy with the one to open documents. The one to create a new document is more like a page add. Here I'd stick with the yellow/orange star instead of the green plus.
The one to save is definitely not what I'd expect to save my document. Additionally it is especially problematic with the upcoming collaboration feature. Users could mistake it for downloading any changes (even if that not is a thing actually). Here a floppy would be preferable.

**Comment by Fatih20** [comment]
> So I'm happy with the one to open documents. The one to create a new document is more like a page add. Here I'd stick with the yellow/orange star instead of the green plus.
> The one to save is definitely not what I'd expect to save my document. Additionally it is especially problematic with the upcoming collaboration feature. Users could mistake it for downloading any changes (even if that not is a thing actually). Here a floppy would be preferable.

With all due respect Mr. Huber, I don't think your solution for the new document isn't ideal and I will propose a counterargument and my solution.

1. A page with a star is uncommon is most other software. Furthermore, a star isn't correlated with "adding stuff" in on itself.
2. A page with a star as in an actual star would be like adding bookmark or favoriting instead. Using this * is not aesthetically pleasing.
3. It's common sense that making new document is clustered with save and open document. This is true from Microsoft Office to MyPaint.
4. Xournal is designed in such a way that when you hover over a button it will show what those button would do.
 
That being said, I do understand your concern about them being too similar. I have a solution of my own.

image

**Comment by LittleHuba** [comment]
The plus is in my opinion adding to something. A new document is not adding to anything. It is a fresh start.

> image

I'm happy with this one.

**Comment by Fatih20** [comment]
Just needed to cleanup all the files.

**Comment by Fatih20** [comment]
Is there any reason why `ui/iconsColor/hicolor/24x24/actions/pageSpinner.png` exists?

**Comment by Technius** [comment]
> Is there any reason why `ui/iconsColor/hicolor/24x24/actions/pageSpinner.png` exists?

It is the icon shown for the page number display in the `Customize Toolbars` dialog:
image

**Comment by Fatih20** [comment]
> > Is there any reason why `ui/iconsColor/hicolor/24x24/actions/pageSpinner.png` exists?
> 
> It is the icon shown for the page number display in the `Customize Toolbars` dialog:
> image

Any reason why is it in png?

**Comment by Fatih20** [comment]
I think I'll be porting it to an svg.

**Comment by Fatih20** [comment]
You can merge this now. They're ready for prime time!

File: ui/iconsColor/hicolor/24x24/actions/page-spinner.svg [hunk 1]
```diff
@@ -0,0 +1,187 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (LittleHuba) [inline ui/iconsColor/hicolor/24x24/actions/page-spinner.svg:1]: Move this file to `ui/iconsColor/hicolor/scalable/actions/page-spinner.svg`

File: ui/iconsMono/hicolor/24x24/actions/page-spinner.svg [hunk 2]
```diff
@@ -0,0 +1,187 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (LittleHuba) [inline ui/iconsMono/hicolor/24x24/actions/page-spinner.svg:1]: Move this file to `ui/iconsMono/hicolor/scalable/actions/page-spinner.svg

**Comment by LittleHuba** [comment]
So GTK is adamant that we shall not use own icons that would override system icons. Only workaround I found so far is renaming them to symbols not provided by the system. But that is somewhat imperfect, as we do not know what symbols OS themes bring with them. Of course the lacking documentation does not help at all. One more reason why I dont like GTK...

**Comment by Fatih20** [comment]
> So GTK is adamant that we shall not use own icons that would override system icons. Only workaround I found so far is renaming them to symbols not provided by the system. But that is somewhat imperfect, as we do not know what symbols OS themes bring with them. Of course the lacking documentation does not help at all. One more reason why I dont like GTK...

Isn't that the same solution you use for the playback icons? It worked before.

**Comment by LittleHuba** [comment]
Best would be to just prefix any icon with `xopp-`.

**Comment by rolandlo** [comment]
@Fatih20 May I ask about your plans of finishing this PR? Would be nice to have it ready for merging.

**Comment by rolandlo** [comment]
@Fatih20 Having your new icons merged would be a great plus when releasing version 1.1.0, since it will look nice and fresh. Do you need help finishing this PR?

**Comment by Fatih20** [comment]
> @Fatih20 Having your new icons merged would be a great plus when releasing version 1.1.0, since it will look nice and fresh. Do you need help finishing this PR?
The icon is done. My part here is finished.

From my understanding, the rest now lie on @LittleHuba and others to make a UI to switch icons between monochrome and colored, among other things.

**Comment by rolandlo** [comment]
@LittleHuba @Technius How do we proceed from here? Do you need some help? @Fatih20 has completed his part, so it is on us to finish this off.

**Comment by LittleHuba** [comment]
@Febbe is the new application initialization finished? That was the blocker for this PR. 

I already tried some ideas on how to enable switching of themes. We can't use the current system as it only supports the light and dark version. I haven't got it running so far.

We also need to revisit the PR for the new application icon. I had to revert it as Windows releases did not show any icon at all after merging it. I suspect a missing file there. @rolandlo would you mind checking this out?

**Comment by rolandlo** [comment]
@LittleHuba Yes, you reverted it, but then a new version of the PR got merged, see #2557. I can check if it works on Windows.

**Comment by LittleHuba** [comment]
Then it should be good. People would have complained by now.

**Comment by rolandlo** [comment]
I have checked it nonetheless. Xournal++ (latest nightly release) works with the new icons. The new icon is correctly displayed inside the app (as seen in the `About` dialog). It is also correctly displayed in the taskbar while running. The icon displayed in the Windows Start menu was the old one, however. It is also used if you right click the executable and attach to the taskbar. 

As I have noticed, the `xournalpp.ico` file (in the `windows-setup` folder) has not been updated. It still shows the old icon. We must fix that. I guess we simply convert the svg-file of the icon into .ico format using a tool like `convert`from ImageMagick.

**Comment by LittleHuba** [comment]
Nice catch @rolandlo. Could you modify the build script to do this for us automatically? This might be the best approach as long as we can use some pre-existing or easy-to-install dependency for this. I would like to remove the redundancy for this situation.

**Comment by rolandlo** [comment]
@LittleHuba Ok, I will modify the build scripts, also accounting for the missing plugin and ressource folders. Just give me some time.

**Comment by Technius** [comment]
Hi @Fatih20, can you clarify what license you are contributing these icons under? The code must be under GPLv2, but the icons are your original contribution. Based on the image metadata, it seems to be some variant of Creative Commons? Need to know for #2148.

**Comment by Fatih20** [comment]
> Hi @Fatih20, can you clarify what license you are contributing these icons under? The code must be under GPLv2, but the icons are your original contribution. Based on the image metadata, it seems to be some variant of Creative Commons? Need to know for #2148.

Yes I'm contributing it under GPLv2. Do I need to change the metadata?

**Comment by Technius** [comment]
> Do I need to change the metadata?

Please do so if it's easy. If it's too troublesome, we can add the metadata info for you.

**Comment by Fatih20** [comment]
> > Do I need to change the metadata?
> 
> Please do so if it's easy. If it's too troublesome, we can add the metadata info for you.

You can add them yourself. Thanks.

**Comment by rolandlo** [comment]
@LittleHuba Any progress on enabling theme switching?

**Comment by Technius** [comment]
@rolandlo I can implement the theme switching stuff if no one else has time for it. For now, I think we should merge this into the `new-icons` branch and deal with the code in a follow-up PR.

**Comment by rolandlo** [comment]
> @rolandlo I can implement the theme switching stuff if no one else has time for it. For now, I think we should merge this into the `new-icons` branch and deal with the code in a follow-up PR. (need to resolve the merge conflicts by using the new icons though).

I would appreciate that.

**Comment by mjg** [comment]
Just a note on the relation to #2795: Over there I suggested similar icon renames but dropped them now. I do introduce a new icon `sidebar-layerstack` (named to match the new name `sidebar-layer` here) copied from the existing (legacy) `layer` icon with an added arrow. The layerstack sidebar view displays the layer stack progressively (up to a certain layer) to match the new export option. It would be great if you introduce a stacked/progressive version of your new `sidebar-layer` icon, too, and name it `sidebar-layerstack`.

**Comment by LittleHuba** [comment]
> @rolandlo I can implement the theme switching stuff if no one else has time for it. For now, I think we should merge this into the `new-icons` branch and deal with the code in a follow-up PR. (need to resolve the merge conflicts by using the new icons though).

@Technius please do take over.

**Comment by Technius** [comment]
Oops, let me see if I can cherry pick the commits on to another branch... It got out of sync and became hard to merge.

**Comment by Technius** [comment]
See #2886 for follow up PR.

**Pull request closed without merging**